data-structure/hashTable.ts

```diff
@@ -0,0 +1,104 @@
+// Read about hash tables and its working mechanism and implement it functionality.
+// NOTE: get acquainted with possible collisions when working with hash tables, take this information into account when implementing.
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+type Item = string | number;
+
+type List = Array<Array<string | Item>>;
+
+type Store = Array<List>;
+
+class HashTable {
+  private readonly store: Store;
+
+  private readonly size: number;
+
+  constructor(size = 13) {
+    this.size = size;
+    this.store = [];
+  }
+
+  add(key: string, value: Item) {
+    const index: number = this.hash(key);
+
+    if (!this.store[index]) {
+      this.store[index] = [[key, value]];
+    } else {
+      const list: List = this.store[index];
+      const matchingIndex: number | void = HashTable.findMatchingIndex(
+        list,
+        key
+      );
+
+      if (matchingIndex) {
+        list[matchingIndex] = [key, value];
+        return;
+      }
+
+      list.push([key, value]);
+    }
+  }
+
+  remove(key: string) {
+    const index: number = this.hash(key);
+    if (this.store[index]) {
+      const list = this.store[index];
+      const matchingIndex: number | void = HashTable.findMatchingIndex(
+        list,
+        key
+      );
+
+      if (matchingIndex || matchingIndex === 0) {
+        this.store[index] = [
+          ...list.slice(0, matchingIndex),
+          ...list.slice(matchingIndex + 1),
+        ];
+      }
+    }
+  }
+
+  find(key: string) {
+    const index = this.hash(key);
+    if (this.store[index]) {
+      const list = this.store[index];
+      const matchingIndex: number | void = HashTable.findMatchingIndex(
+        list,
+        key
+      );
+      if (matchingIndex || matchingIndex === 0) return list[matchingIndex];
+    }
+    return "No matches";
+  }
+
+  hash(value: string) {
+    let index: number;
+    index = 0;
```

Comment:
combine 75 and 76 lines. and you don't need to define type here. ts will take this variable as a number by default because you define the value